src/node_watchdog.h

```diff
@@ -55,24 +61,50 @@ class Watchdog {
   bool* timed_out_;
 };
 
-class SigintWatchdog {
+class SigintWatchdogBase {
+ public:
+  virtual ~SigintWatchdogBase() = default;
+  virtual SignalPropagation HandleSigint() = 0;
+};
+
+class SigintWatchdog: public SigintWatchdogBase {
  public:
   explicit SigintWatchdog(v8::Isolate* isolate,
                           bool* received_signal = nullptr);
   ~SigintWatchdog();
   v8::Isolate* isolate() { return isolate_; }
-  void HandleSigint();
+  SignalPropagation HandleSigint() override;
 
  private:
   v8::Isolate* isolate_;
   bool* received_signal_;
 };
 
+class JsSigintWatchdog : public HandleWrap, public SigintWatchdogBase {
```

Comment:
Just curious, was switching the bases necessary for the postmortem metadata test? That would probably point to a design problem with the postmortem metadata.

Reply:
No, I just found existing code putting HandleWrap first so I would like keep it consistent since there is no strong opinion about it. 

The postmortem metadata test was affected by handles initiated during`pre_execution`. Actually if I got it right, case `DebugSymbolsTest, HandleWrapList` was expected to find the last item of the handle_wrap_queue for the newly initiated handle wrap instance.

Reply:
Okay, thanks for clarifying that – the order of the classes should not make a difference :)